llm/prompt.go

```diff
@@ -68,27 +68,58 @@ type Response struct {
 }
 
 func getProvider(ctx context.Context) LLMProvider {
-	if key := os.Getenv("GEMINI_API_KEY"); key != "" {
-		provider := NewGeminiProvider(key, ctx)
-		if provider != nil {
-			return provider
-		}
+	// Define provider configurations
+	providerConfigs := map[string]struct {
```

Comment:
This can be extracted outside of the func and also simplified to map name->func as we can embed lookup inside the provider creation